pkg/util/file_indexer.go

```diff
@@ -83,6 +85,60 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the context
+func gitIgnoreFilePath(directory string) (string, error) {
+	_, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(directory, ".gitignore"), nil
+}
+
+// addOdoFileIndex adds odo-file-index.json to .gitignore
+func addOdoFileIndex(ignoreFile string) error {
+	var data []byte
+	file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_RDWR, 0600)
+	if err != nil {
+		return errors.Wrap(err, "failed to open .gitignore file")
+	}
+	defer file.Close()
+
+	if data, err = ioutil.ReadFile(ignoreFile); err != nil {
+		return errors.Wrap(err, "failed reading data from .gitignore file")
+	}
+	// check whether .odo/odo-file-index.json is already in the .gitignore file
+	if !strings.Contains(string(data), ".odo/odo-file-index.json") {
+		if _, err := file.WriteString("\n" + ".odo/odo-file-index.json"); err != nil {
+			return errors.Wrap(err, "failed to Add odo-file-index.json to .gitignore file")
+		}
+	}
+	return nil
+}
+
+// Check .gitignore file exists or not
+func checkGitIgnoreFile(directory string) error {
+
+	ignoreFile, err := gitIgnoreFilePath(directory)
+	_, err = os.Stat(ignoreFile)
+	files, _ := ioutil.ReadDir(filepath.Join(directory))
+
+	if len(files) > 1 {
+		if err != nil {
```

Comment:
Which `err` this checking for? I think it's the one in the line `_, err = os.Stat(ignoreFile)`. What's the point in checking for that err here?

Reply:
https://github.com/openshift/odo/pull/2305#discussion_r342974192